DashWallet/Sources/Models/CrowdNode/CrowdNode.swift

```diff
@@ -763,16 +805,41 @@ extension CrowdNode {
             }
         }
     }
+    
+    private func checkIfEmailRegistered(address: String) {
+        Task {
+            let usingDummyEmail = await webService.isDefaultEmail(address: address)
+            
+            if usingDummyEmail {
+                // User email isn't set yet. Check the message status in case there is an error
+                let messageId = signedEmailMessageId
+                
+                if messageId != -1 {
+                    let message = await webService.checkMessageStatus(id: messageId, address: address)
+                    
+                    if message?.messageStatus.lowercased() == kMessageFailedStatus {
+                        // Operation failed
+                        apiError = CrowdNode.Error.messageStatus(error: message?.result ?? "")
+                        signedEmailMessageId = -1
+                        changeOnlineState(to: .none)
+                    }
+                }
+            } else {
+                // Good to go
+                changeOnlineState(to: .signingUp)
+            }
+        }
+    }
 
-    private func waitForApiAddressConfirmation(primaryAddress: String, apiAddress: String) async -> DSTransaction {
+    private func waitForApiAddressConfirmation(primaryAddress: String, apiAddress: String) async -> (tx: DSTransaction, didWait: Bool) {
```

Comment:
@Syn-McJ I'm not sure I understand this change, why do we need to return a flag from async method that shows whether it was invoked synchronously or not in order to check availability of confirmation. I think, it's better to keep it as it was and has some additional flag that tells whether we need to invoke `notifyIfNeeded`.

Reply:
@tikhop the problem is that after the wallet reset / the network is changed, we don't have a good way to determine if we need to show this notification or not. Any other flag that we keep will be erased.

The boolean here isn't about how the method was invoked, but whether we had to wait for the confirmation or not. If we waited - means the confirmation was still in process and we need to inform the user that it's finished. If not - means the confirmation was already done and the user has already received the notification.

Reply:
@Syn-McJ I would keep it simple and use the flag instead. If user resets the wallet, the flag is also reseted then it's fine to show notification once again.

Reply:
Showing the notification again is the issue that was raised during QA and that we're trying to fix :) 
Specifically, the user would see "Your account is confirmed" every time the network is changed.

Reply:
I can probably do this with a flag but in a slightly different way.

Reply:
Okay, you have one flag for all networks and if I try to create on mainnet and then switch on testnet you need to reset the flag as well, otherwise you can't display a notification if we created an account once on a different network. We can keep it as it is, but I think a better solution is to have a different flag per network since you can have two accounts with different networks.

Reply:
This doesn't sound like "keep it simple" to me :)
Anyway, I'll do the flag solution and trigger it when the user initiates the confirmation. Then we'll only show the notification if the flag is set.

Reply:
Fixed